amm/drink-tests/src/router_tests.rs

```diff
@@ -74,9 +74,214 @@ fn add_liquidity(mut session: Session) {
+
+    let (_a, _b, liquidity_minted) = router::add_liquidity(
+        &mut session,
+        router.into(),
+        ice.into(),
+        wood.into(),
+        token_amount,
+        token_amount,
+        bob(),
+    );
+
+    let ice_wood_pair: pair_contract::Instance =
+        factory::get_pair(&mut session, factory.into(), ice.into(), wood.into());
+
+    let bob_lp_balance: u128 = psp22::balance_of(&mut session, ice_wood_pair.into(), bob());
+    assert_eq!(liquidity_minted, bob_lp_balance);
+
+    let swap_amount = 10_000 * 10u128.pow(18);
+    // 0.3% is would be 30 tokens exactly but due to rounding we can lose up to 1 dust.
+    let swap_res = session
+        .execute(router.swap_exact_tokens_for_tokens(
+            swap_amount,
+            0,
+            vec![ice.into(), wood.into()],
+            bob(),
+            deadline,
+        ))
+        .unwrap()
+        .result
+        .unwrap()
+        .unwrap();
+
+    assert_eq!(swap_res[0], swap_amount);
+    // We cannot assert that as it depends on the pool's liquidity and swap amount. The price will move from the spot price.
+    // assert!(swap_res[1] >= min_amount_out);
+
+    // No fees distributed until the burn/mint transaction.
+    assert!(psp22::balance_of(&mut session, ice_wood_pair.into(), charlie()) == 0);
+    // Burn some liquidity to trigger fee collection.
+    psp22::increase_allowance(
+        &mut session,
+        ice_wood_pair.into(),
+        router.into(),
+        liquidity_minted,
+        BOB,
+    )
+    .unwrap();
+    let (_ice, _wood) = session
+        .execute(router.remove_liquidity(
+            ice.into(),
+            wood.into(),
+            liquidity_minted,
+            0,
+            0,
+            bob(),
+            deadline,
+        ))
+        .unwrap()
+        .result
+        .unwrap()
+        .unwrap();
+
+    // Fees now sent to `fee_to` address (CHARLIE).
+    let charlie_lp_balance = psp22::balance_of(&mut session, ice_wood_pair.into(), charlie());
+    // Trading fee is 0.3%.
+    let trading_fee = 0.3 / 100.0;
+    // Protocol fees are 1/6 of the trading fees.
+    let expected = (swap_amount as f64 * trading_fee) / 6.0;
+    // Charlie withdraws his fees
+    psp22::increase_allowance(
+        &mut session,
+        ice_wood_pair.into(),
+        router.into(),
+        charlie_lp_balance,
+        CHARLIE,
+    )
+    .unwrap();
+    let (protocol_fees_ice, protocol_fees_wood) = router::remove_liquidity(
+        &mut session,
+        router.into(),
+        ice.into(),
+        wood.into(),
+        charlie_lp_balance,
+        0,
+        0,
+        charlie(),
+    );
+    // We cannot assert exactly how much fees Charlie will get, due to roundings etc,
+    // but it should be close to the expected value
+    let protocol_fees = protocol_fees_ice + protocol_fees_wood;
+    let percentile = expected * 9999.0 / 10000.0;
+    assert!(protocol_fees >= expected as u128 - percentile as u128);
```

Comment:
This does not look good, you probably wanted something like 
`protocol_fees >= ((expected as f64)*0.9999) as u128`

Reply:
Ouch. Definitely.

Reply:
Please have a look at [Router test](https://github.com/Cardinal-Cryptography/common-amm/pull/86/commits/26bd7511abaaeb0170bcd4fbb5c9e2fadd2b2811).